packages/vscode-extension/src/project/metro.ts

```diff
@@ -292,6 +292,82 @@ export class Metro implements Disposable {
     ws.close();
   }
 
+  private async evaluateJsFromCdp(
+    webSocketDebuggerUrl: string,
+    source: string,
+    timeoutMs: number = 2000
+  ): Promise<string | undefined> {
+    const REQUEST_ID = 0;
+    let timeoutHandle: NodeJS.Timeout;
+
+    const wsUrl = new URL(webSocketDebuggerUrl);
+    wsUrl.hostname = "localhost";
+    wsUrl.port = String(this._port);
+
+    return new Promise((resolve, reject) => {
+      let settled = false;
+      const ws = new WebSocket(wsUrl);
+
+      timeoutHandle = setTimeout(() => {
+        Logger.debug(`[evaluateJsFromCdpAsync] Request timeout from ${wsUrl.toString()}`);
+        reject(new Error("Request timeout"));
+        settled = true;
```

Comment:
That sounds redundant. Shouldn't you just close the websocket, and the `on('close')` with settled will decide if it's "settled" or not?

Reply:
no if we did that we wouldn't know the close reason in on(close) and we want to log that